### ISSUE TITLE
Improve accessibility for screen readers

### ISSUE THREAD

**Issue by Silmathoron** [iv-org/invidious#1311]
Hi there,
I really appreciate the work done on Invidious.
Unfortunately, when I recommended it to someone using a screen reader, they told me that it was not accessible enough for them to use it.
Apparently, issues notably include:

* Empty links
* Unlabeled buttons

Would it be possible to see how to enhance the interface on that side?

**Comment by rkingett** [comment]
Using [NVDA, a free and open source screen reader](http://www.ninite.com/nvda/) on Windows, checking out the website, the following problems occur.

1. Empty links. Honestly, there's so many of them I can't tell you what they are, but they are all over the place. If it was just a few, I could tell you which links are empty, but they appear to be in the site header and many empty links on video pages.
2. keyboard focus is poor when interacting with the video player region. Without enabling focus mode on NVDA, the video does not play.
3. Many links are broken up into 2 links that lead to the same place. 2 separate links lead to the same target. For example, StarTalk is one link.
405K views is another link, but they both lead to the same video. Could we make those 2 links into one long link?
3. Using [HTML Semantic Elements](https://www.w3schools.com/html/html5_semantic_elements.asp) would greatly improve navigation. ARIA rolls are not needed here because HTML is better, overall. I propose we use, MAIN, FOOTER, NAV, and SUMMARY, 
4. After selecting the audio version, It is still showing the video player? I can't find the audio player. Or am I missing something?

I haven't tried the IOS app yet, if there is one, but I'd be happy to answer questions.

**Comment by SamantazFox** [comment]
Hello there! Sorry for the lack of replies, a lot has been going on in this project. In essence: the original dev left in sptember 2020, a new team of volunteers have taken back the project, and I'm now the only major developer working on Invidious.

Screen reader acessiblity is being taken into account, it's just a bit slow because it's done along with the user interface rewrite, which takes time (plus, many other parts of the code are being rewritten too to ease maintenance and bug fixing).
